cmd/kubeadm/app/apis/kubeadm/v1beta1/types.go

```diff
@@ -106,12 +106,23 @@ type ClusterConfiguration struct {
 	// CertificatesDir specifies where to store or look for all required certificates.
 	CertificatesDir string `json:"certificatesDir"`
 
-	// ImageRepository what container registry to pull control plane images from
+	// ImageRepository what container registry to pull control plane images from.
+	// If empty, `k8s.gcr.io` will be used by default except in case of kubernetes version is a CI build (kubernetes version starts with `ci/` or `ci-cross/`)
+	// when `gcr.io/kubernetes-ci-images` will be used as a default for control plane components and for kube-proxy, while `k8s.gcr.io`
+	// will be used for all the other images.
 	ImageRepository string `json:"imageRepository"`
 	// UnifiedControlPlaneImage specifies if a specific container image should
-	// be used for all control plane components.
+	// be used for all control plane components and for kube-proxy.
+	// In case this value is set, kubeadm does not change automatically the version of the above components during upgrades.
 	UnifiedControlPlaneImage string `json:"unifiedControlPlaneImage"`
 
+	// DNS defines the options for the DNS add-on installed in the cluster.
+	DNS DNS `json:"dns"`
+
+	// Proxy defines the options for the proxier add-on installed in the cluster.
+	// Configurability of the proxy can be achieved via component config as well.
+	Proxy Proxy `json:"dns"`
```

Comment:
my original idea was to have these under a parent Addons sub-struct or a list, but this is fine too.
...unless if we decide at some point to add a bunch more core addons, which will then leave all of them floating in the root config.